### DISCUSSION TITLE
Check visible element

### DISCUSSION THREAD

**Discussion by tit** [cypress-io/cypress#17203] [Questions and Help] [answered]
Hi, i have some html code (for example):

```
<button onClick="showItems()">Show Items</button>

<div class="item" style="display: none">foo</div>
<div class="item" style="display: none">bar</div>
```

I want check, what after clicking on the button, then `<div>` is visible (do not have `display: none` style)

I see two variant:
1. `.should('be.visible')`
2. `.should('not.have.css', 'display', 'none')`

What the difference between `.should('be.visible')` and `.should('not.have.css', 'display', 'none')`?

**Comment by jennifer-shehane** [comment, marked as answer]
These assertions come from chai-jquery. I believe `is.visible` will pass if it would be queryable via `:visible` css, which could be caused by other css styles. Assertions `.not.have.css` will be more specific - that tthat specific sttyle  does not exist on that element.

**Reply by tit** [reply]
Thank you, @jennifer-shehane
